plugins/dbviewer/frontend/public/javascripts/countly.models.js

```diff
@@ -74,19 +74,38 @@
         });
     };
 
-    countlyDBviewer.executeAggregation = function(db, collection, aggregation, callback) {
-        return $.ajax({
-            type: "GET",
-            url: countlyCommon.API_URL + "/o/db",
-            data: {
-                dbs: db,
-                collection: collection,
-                aggregation: aggregation
-            },
-            success: function(json) {
+    countlyDBviewer.executeAggregation = function(db, collection, aggregation, app_id, task_id, callback) {
+        if (task_id) {
+            return $.when(countlyTaskManager.fetchResult(task_id, function(json) {
                 callback(json);
-            }
-        });
+            }));
+        }
+        else {
+            return $.ajax({
+                type: "GET",
+                url: countlyCommon.API_URL + "/o/db",
+                data: {
+                    dbs: db,
+                    collection: collection,
+                    aggregation: aggregation,
+                    app_id: app_id
+                },
+                success: function(json) {
+                    if (json.aaData && json.aaData.task_id) {
+                        app.recordEvent({
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['app' is not defined. (no-undef)](https://app.codacy.com/app/ar2rsawseen/countly-server/pullRequest?prid=4534064)

Reply:
Need to define app as global in this file